frontend/test/metabase/scenarios/collections/bookmarks.cy.spec.js

```diff
@@ -0,0 +1,92 @@
+import _ from "underscore";
+import { restore, sidebar } from "__support__/e2e/cypress";
+
+describe("Bookmarks in a collection page", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("cannot add bookmark to root collection", () => {
+    cy.visit("/collection/root");
+
+    cy.icon("bookmark").should("not.exist");
+  });
+
+  it("can add and remove bookmarks from collection from its page", () => {
+    cy.visit("/collection/1");
+
+    // Add bookmark
+    cy.icon("bookmark").click();
+
+    // Bookmark Icon should be blue once Collection is bookmarked
+    cy.icon("bookmark")
+      .should("have.css", "color")
+      .and("eq", "rgb(80, 158, 227)");
+
+    sidebar().within(() => {
+      cy.findByText("Bookmarks");
+      cy.findByText("Bobby Tables's Personal Collection");
+
+      // Once there is a list of bookmarks,
+      // we add a heading to the list of collections below the list of bookmarks
+      cy.findByText("Collections");
+    });
+
+    // Remove bookmark
+    cy.findByTestId("collection-menu").within(() => {
+      cy.icon("bookmark").click();
+    });
+
+    sidebar().within(() => {
+      cy.findByText("Bobby Tables's Personal Collection").should("not.exist");
+      cy.findByText("Bookmarks").should("not.exist");
+
+      // Once there is no list of bookmarks,
+      // we remove the heading for the list of collections
+      cy.findByText("Collections").should("not.exist");
+    });
+
+    // Bookmark Icon should be dark gray once bookmark is removed from Collection
+    cy.icon("bookmark")
+      .should("have.css", "color")
+      .and("eq", "rgb(76, 87, 115)");
```

Comment:
Same comment about color assertions as for the check above

Reply:
@kulyk makes 100% sense to me.

I went to unit tests and after quite some time, couldn't get styled colors from the svg.
All the basic ways of retrieving them failed.

So to be able to move forward, I've removed tests on colors.

Reply:
Thanks! For SVGs, you'd probably need to assert the `fill` property. On the other hand, I wonder if it'd be better to maybe add a visual Percy test for the whole collections page? We can catch a lof of visual regressions in this way: various bookmarks state, official collections, the collection contents layout, etc.

Reply:
Yes @kulyk, I just added a visual regression test. Thank you.